x/pairing/client/cli/tx_stake_provider.go

```diff
@@ -365,3 +400,58 @@ func getValidator(clientCtx client.Context, provider string) string {
 	}
 	return validatorBiggest.OperatorAddress
 }
+
+// CreateGrantFeeMsg cosntructs a feegrant GrantAllowance msg for specific TXs to allow the provider use the vault's funds for gas fees
+func CreateGrantFeeMsg(granter string, grantee string) *feegrant.MsgGrantAllowance {
+	if grantee == granter {
+		// no need to grant allowance if the granter and grantee are the same (vault = provider)
+		return nil
+	}
+	granterAcc, err := sdk.AccAddressFromBech32(granter)
+	if err != nil {
+		utils.LavaFormatError("failed granting feegrant for gas fees for granter", err,
+			utils.LogAttr("granter", granter),
+		)
+		return nil
+	}
+
+	granteeAcc, err := sdk.AccAddressFromBech32(grantee)
+	if err != nil {
+		utils.LavaFormatError("failed granting feegrant for gas fees for grantee", err,
+			utils.LogAttr("grantee", grantee),
+		)
+		return nil
+	}
+
+	grant, err := feegrant.NewAllowedMsgAllowance(&feegrant.BasicAllowance{}, []string{
+		// pairing module TXs
+		sdk.MsgTypeURL(&types.MsgRelayPayment{}),
+		sdk.MsgTypeURL(&types.MsgStakeProvider{}), // for modify-provider TX
+		sdk.MsgTypeURL(&types.MsgFreezeProvider{}),
+		sdk.MsgTypeURL(&types.MsgUnfreezeProvider{}),
+
+		// conflict module TXs
+		sdk.MsgTypeURL(&conflicttypes.MsgDetection{}),
```

Comment:
detection is a consumer

Reply:
done. see https://github.com/lavanet/lava/commit/165e856d35ac919bd67830cb66b0ccc4f8b156e6